### DISCUSSION TITLE
Rust GameDev Community Assets Store

### DISCUSSION THREAD

**Discussion by voxelias** [rust-gamedev/wg#113] [Ideas]
Hello everyone!

I've came up with an idea about creation of the Assets Store for Rust GameDev projects. After a quick brainstorming with @AngelOnFira we agreed on sharing it in the work group.

Long story short:

1. Create a resource (website) where users can upload and share assets (models, textures, sounds, code, plugins etc) they create.
2. Game developers can download these assets manually or integrate their projects with the store using some API. A plugin can be prepared and shared on [crates.io](https://crates.io). The integration can open some additional features, like live updates (lets say if an artist fixes some issue in a model and uploads new version).

## Why dedicated Rust GameDev Assets Store

1. Better user experience: The Store can motivate artists to submit assets in preferred supported by rest of the Rust projects formats (gltf, png, ogg etc) and/or with optional info like `Works on Amethyst, Bevy, Dotrix`
2. It may attract new game developers to the community due to `Batteries-Included` concept.
3. Can be used to attract some investments into existing projects (more info bellow)

## Why single store instead of store per engine

1. This will save a lot of developers capacities
2. It will be easier to maintain with own community

## Artists benefits

1. An artist can share assets for free, but with info that more can become available with lets say patreon subscription, so he/she can attract more sponsors 
2. An artist can sell assets for money directly from the Store

## Developer's benefits

1. A store of compatible assets itself
2. Possibility to request an asset or find an artist for a project
3. The Store can offer a feature to artists to share their income with chosen gamedev projects.

## MVP

- Artists can create a profile and share their assets under CC-0
- Developers can download assets from web or using API

## Questions
- Would you be interested to participate in development of such Store?
- Would you be interested to use such Store?
- Do you need more info? Feel free to ask a question.

Please, share your thoughts! 

Cheers!

**Comment by AngelOnFira** [comment]
Here were our full discussion points (although much of the context is missing): https://hackmd.io/6mDghKnzTnCUTZibbFwj4A

**Comment by PsichiX** [comment]
why not just partner with Itch.io? 🤔 
the solid infrastructure is already there, what's left is to make a common API for game engines and stuff to ease interacting with free/bought assets and store itself.

**Comment by Ralith** [comment]
Similar storefronts require full-time staffing to operate. Is someone volunteering to invest that amount of effort indefinitely (if at all)? If not, then discussion is moot. Historically the WG hasn't had resources to address even much smaller problems.

**Comment by alice-i-cecile** [comment]
I don't think it makes sense across engine boundaries:

- game logic crates will almost never work across engines. If they do, they can live on crates.io with the appropriate tag.
- game templates will never work across engines.
- editor extensions will never work across engines.
- raw assets aren't Rust-specific: these can live on itch.io happily for more visibility and simplicity.
- complex assets (scenes, asset pipelines) are game-engine specific.

As an engine, you lose a huge amount of control over the features, financial model and branding, in exchange for lower dev and organizational costs (maybe).

**Comment by cart** [comment]
(Bias warning: I am the lead Bevy developer and we already have our own "asset store" ... however it is still in its very early stages)

In my opinion the big draw of asset stores is a centralized place containing things that are all compatible with a specific type of project (and each other). Even if you believe in a "multi-engine asset store" (something like turbosquid?),  I'm not sure theres much justifying a "rust asset store" over a "generic asset store". rg3d "assets" won't be any more compatible with bevy "assets" than godot "assets". If thats the market you're aiming for, I see no reason to make it rust specific.

I think there is definitely value in a "generic rust gamedev crate listing", but we already have those in the form of the [crates.io game-development category](https://crates.io/categories/game-development) and [arewegameyet.rs](https://arewegameyet.rs/).

As Bevy is specifically called out in the "Why would Bevy want to use this rather than making their own", I'll share my thoughts on why I wouldn't use this:

1. I want a centralized listing that only contains things compatible with Bevy. We have our own "asset format" that we will expand over time. We intend to have deep integrations with the asset store (for things like asset vs crate deployment, bevy version support, distinguishing between "bevy scenes" and "bevy crates", editor integration, etc). I don't want Bevy developers to need to learn anything but the "bevy asset format".
2. I want control over branding and presentation. Notice that [Bevy Assets](https://bevyengine.org/assets/) has the same look and feel as [The Bevy Book](https://bevyengine.org/learn/book/introduction/). We've spent a lot of time and effort making everything feel integrated and I don't want to give that up.
3. Funding open source is hard. There is no way we can build a full-time market-rate "Bevy dev team" on sponsorships alone. An asset store is one of the more viable ways to fund an open source game engine (godot is currently exploring this area). I have no interest in handing over governance and financial control of the Bevy Asset store to a non-Bevy entity.
